thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/VcardParser.java

```diff
@@ -0,0 +1,543 @@
+                    try {
+                        writeExtractedImage(extractedFilePath, data);
+                        derivedFilesCreated.add(fileManager.addDerivedFile(extractedFileName, getFileRelativePath(parentFileName, extractedFileName), data.length,
+                                abstractFile.getCtime(), abstractFile.getCrtime(), abstractFile.getAtime(), abstractFile.getAtime(),
+                                true, abstractFile, null, EmailParserModuleFactory.getModuleName(), null, null, TskData.EncodingType.NONE));
+                    } catch (IOException | TskCoreException ex) {
+                        logger.log(Level.WARNING, String.format("Could not write image to '%s' (id=%d).", extractedFilePath, abstractFile.getId()), ex); //NON-NLS
+                    }
+                }
+                if (!derivedFilesCreated.isEmpty()) {
+                    services.fireModuleContentEvent(new ModuleContentEvent(abstractFile));
+                    context.addFilesToJob(derivedFilesCreated);
+                }
+            }
+            else {
+                logger.log(Level.INFO, String.format("Skipping photo extraction for file '%s' (id=%d), because it has already been processed.",
+                        abstractFile.getName(), abstractFile.getId())); //NON-NLS
+            }
+        } catch (SecurityException ex) {
+            logger.log(Level.WARNING, String.format("Could not create extraction folder for '%s' (id=%d).", parentFileName, abstractFile.getId()));
+        }
+    }
+    
+    /**
+     * Writes image to the module output location.
+     *
+     * @param outputPath Path where images is written.
+     * @param data       Byte representation of the data to be written to the
+     *                   specified location.
+     */
+    private void writeExtractedImage(String outputPath, byte[] data) throws IOException {
+        File outputFile = new File(outputPath);
+        FileOutputStream outputStream = new FileOutputStream(outputFile);
+        outputStream.write(data);
+    }
+    
+    /**
+     * Creates a unique name for a file by concatentating the file name and the
+     * file object id.
+     *
+     * @param file The file.
+     *
+     * @return The unique file name.
+     */
+    private String getUniqueName(AbstractFile file) {
+        return file.getName() + "_" + file.getId();
+    }
+    
+    /**
+     * Gets the relative path to the file. The path is relative to the case
+     * folder.
+     *
+     * @param fileName Name of the the file for which the path is to be
+     *                 generated.
+     *
+     * @return The relative file path.
+     */
+    private String getFileRelativePath(String parentFileName, String fileName) throws NoCurrentCaseException {
+        // Used explicit FWD slashes to maintain DB consistency across operating systems.
+        return "/" + getRelModuleOutputPath() + "/" + parentFileName + "/" + fileName; //NON-NLS
```

Comment:
Remove the leading slash (it gets stripped off now anyway) and change to use either the file separator or Paths.get() (I changed the place you likely copied this from a few weeks ago https://github.com/sleuthkit/autopsy/pull/4507/files)